model/tensorflow/dffml_model_tensorflow/dnnr.py

```diff
@@ -14,22 +14,25 @@
 from dffml.model.model import Model
 from dffml.accuracy import Accuracy
 from dffml.util.entrypoint import entry_point
-from dffml.base import BaseConfig
+from dffml.base import BaseConfig, config
 from dffml.util.cli.arg import Arg
 from dffml.feature.feature import Feature, Features
 from dffml.util.cli.parser import list_action
 
 from dffml_model_tensorflow.dnnc import TensorflowModelContext
 
 
-@dataclass(init=True, eq=True)
+@config
 class DNNRegressionModelConfig:
-    directory: str
-    steps: int
-    epochs: int
-    hidden: List[int]
+
     predict: str  # feature_name holding target values
     features: Features
+    directory: str = os.path.join(
+        os.path.expanduser("~"), ".cache", "dffml", "tensorflow"
+    )
+    steps: int = 3000
+    epochs: int = 30
+    hidden: List[int] = field(default_factory=lambda: [12, 40, 15])
```

Comment:
Nice! Although, I don't understand the need for `default_factory`?

Reply:
It throws an error other wise these(
[1](https://stackoverflow.com/questions/53632152/why-cant-dataclasses-have-mutable-defaults-in-their-class-attributes-declaratio)[2](https://stackoverflow.com/questions/52063759/passing-default-list-argument-to-dataclasses) ) these links provide a better explanation.

Reply:
Ah yes of course, I forgot the list test that's already present didn't include a default. So we didn't run into this yet. I'd like to preserve the automatic creation of `__hash__` via `Frozen=True`. I take it this makes that a no go?

Reply:
Yeah,also stuff like [this](https://github.com/intel/dffml/blob/a09d12b8c63659df734f7fd2864d799618c9a3a1/model/tensorflow/dffml_model_tensorflow/dnnc.py#L151) :/